spec/jobs/imo/invite_patient_spec.rb

```diff
@@ -24,43 +22,14 @@
         }.to raise_error(ActiveRecord::RecordNotFound)
       end
 
-      it "creates an ImoAuthorization with 'invited' status when invitation is sent" do
-        patient = create(:patient)
-        allow_any_instance_of(ImoApiService).to receive(:invite).and_return(:invited)
-
-        expect {
-          described_class.perform_async(patient.id)
-          described_class.drain
-        }.to change { ImoAuthorization.count }.from(0).to(1)
-        imo_auth = ImoAuthorization.last
-        expect(imo_auth.status).to eq("invited")
-        expect(imo_auth.patient_id).to eq(patient.id)
-      end
-
-      it "creates an ImoAuthorization with 'no_imo_account' status when invited user has no Imo account" do
+      it "invites the patient to Imo" do
         patient = create(:patient)
-        allow_any_instance_of(ImoApiService).to receive(:invite).and_return(:no_imo_account)
+        imo_service = double
```

Comment:
This can be an [instance_double](https://relishapp.com/rspec/rspec-mocks/v/3-10/docs/verifying-doubles/using-an-instance-double), which is safer to use for these sorts of mock based tests:

`imo_double = instance_double(ImoApiService)`

Reply:
I tried:
```
imo_double = instance_double(ImoApiService)
expect(imo_service).to receive(:send_invitation).with(patient)
```
And several variations on that theme. Didn't work for me. Not sure why.

Reply:
oh, you would still have to stub it into place for it to work....something like

```
imo_service = instance_double(ImoApiService)
allow(ImoApiService).to receive(:new).and_return(imo_service)
expect(imo_service).to receive(:something).and_return(:something)
```